### PR TITLE
Add FakePlayer

### PR DESCRIPTION
**Pull request by qouteall**

### CODE REVIEW

**Comment by TheGlitch76** [comment]
Isn't there a factory that is needed to actually construct the player?
Are there any patches that are related to the player?

File: patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayer.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	@Override
+	public void sendMessage(Text component) {
+	}
+
+	@Override
+	public void increaseStat(Stat par1StatBase, int par2) {
+	}
+
+	//@Override public void openGui(Object mod, int modGuiId, World world, int x, int y, int z){}
```
Comment (TheGlitch76) [inline patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayer.java:57]: Why is this commented out?

Comment (qouteall) [reply patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayer.java]: It's copied from forge. removed

File: patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayerFactory.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	 * Mods should either hold weak references to the return value, or listen for a
+	 * WorldEvent.Unload and kill all references to prevent worlds staying in memory.
+	 */
+	public static FakePlayer get(ServerWorld world, GameProfile username) {
+		if (!fakePlayers.containsKey(username)) {
+			FakePlayer fakePlayer = new FakePlayer(world, username);
+			fakePlayers.put(username, fakePlayer);
+		}
+
+		return fakePlayers.get(username);
```
Comment (coderbot16) [inline patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayerFactory.java:59]: AFAIK this should do the same thing
```suggestion
		return fakePlayers.computeIfAbsent(username, $ -> new FakePlayer(world, username));
```

File: patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayerFactory.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	public static void unloadWorld(ServerWorld world) {
+		fakePlayers.entrySet().removeIf(entry -> entry.getValue().world == world);
+
+		// This shouldn't be strictly necessary, but lets be aggressive.
```
Comment (coderbot16) [inline patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayerFactory.java:65]:
```suggestion
		// This shouldn't be strictly necessary, but let's be aggressive.
```

File: patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayer.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.network.packet.c2s.play.ClientSettingsC2SPacket;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.network.ServerPlayerInteractionManager;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.stat.Stat;
+import net.minecraft.text.Text;
+import net.minecraft.util.math.Vec3d;
+
+//Preliminary, simple Fake Player class
```
Comment (coderbot16) [inline patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayer.java:34]:
```suggestion
// Preliminary, simple Fake Player class
```

File: patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayerFactory.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		if (MINECRAFT_PLAYER != null && MINECRAFT_PLAYER.get() != null && MINECRAFT_PLAYER.get().world == world) {
+			FakePlayer mc = MINECRAFT_PLAYER.get();
+
+			if (mc != null && mc.world == world) {
+				MINECRAFT_PLAYER = null;
+			}
+		}
```
Comment (coderbot16) [inline patchwork-extensions/src/main/java/net/minecraftforge/common/util/FakePlayerFactory.java:72]:
```suggestion
		if (MINECRAFT_PLAYER == null) {
			return;
		}

		FakePlayer mc = MINECRAFT_PLAYER.get();

		if (mc != null && mc.world == world) {
			MINECRAFT_PLAYER = null;
		}
```

**Pull request closed without merging**